rundeckapp/src/main/groovy/rundeck/services/audit/AuditEventsService.groovy

```diff
@@ -280,11 +281,12 @@ class AuditEventsService
     /**
      * Extract the username from an authentication object.
      */
-    private static String extractUsername(Authentication authentication) {
-        if (!authentication) {
+    @PackageScope
+    static String extractUsername(Authentication authentication) {
+        if (!authentication || authentication.principal == "") {
```

Comment:
I don't remember the exact reason, but i remebmer there was a reason for which i try to evaluate first authentication.name before authentication.principal.name, which has to do with the principal not always available or usable. This addition to the if would return null if principal is null even if we have auth.name available.